### PR TITLE
feature: access a private Maven repo with gpg in a Leiningen project

### PR DESCRIPTION
**Pull request by blue0513**
Thank you for the amazing library 🎉 
I have a suggestion bellow.

## What

- Allow antq to read `~/.lein/profiles.clj`
- Allow antq to decrypt `credentials.clj.gpg`

## Why

- antq cannot resolve private maven repos listed in `profiles.clj`
- antq cannot decrypt `credentials.clj.gpg` to fetch artifact versions via leiningen

### Reploduction steps

1. Create the following Clojure code and save it as `/path/to/my-project/project.clj`:

```clojure
(defproject dep-test "0.0.1-SNAPSHOT"
  :dependencies [[my/private-lib "0.0.1"]]) ;; latest is `0.0.2`
```

2. Create the following Clojure code and save it as `~/.lein/profiles.clj`:
```clojure
;; Using GPG. ref: https://cljdoc.org/d/leiningen/leiningen/2.9.3/doc/using-gpg
{:user {:repositories [["internal-snapshots" {:url "https://my/private/repository"
                                              :creds :gpg}]]}}
```

3. Execute the following command:
```shell
$ clojure -M:outdated --directory="/path/to/my-project"
```

The output should be as follows:
```shell
| :file                           | :name              | :current | :latest         |
|---------------------------------+--------------------+----------+-----------------|
| /path/to/my-project/project.clj | my/private-lib     | 0.0.1    | Failed to fetch |
```

### CODE REVIEW

**Comment by liquidz** [comment]
@blue0513 Sorry for late reply 🙇 
I'll have a look later.

**Review by liquidz — CHANGES_REQUESTED** [review]
@blue0513 Thanks for your contribution!

File: src/antq/util/leiningen.clj [hunk 1]
```diff
@@ -17,6 +23,36 @@
...
+  [& args]
+  (try
+    (apply shell/sh (gpg-program) args)
+    (catch java.io.IOException e
+      {:exit 1 :err (.getMessage e)})))
+
+(defn- credentials-fn
+  "cf. https://leiningen.org/reference.html"
+  ([] (let [cred-file (io/file (lein-home) "credentials.clj.gpg")]
+        (if (.exists cred-file)
```
[the remark this answers is not in the document]
Comment (blue0513) [reply src/antq/util/leiningen.clj:40]: Ah... and I apologize for forgetting to mention the MOST important point.

These changes in this file heavily reference Leiningen's code.
Leiningen is under the EPL license, so we should take Copyleft into consideration.

But, I believe Antq is already under the EPL license.
So do we not need to take any special precautions?

File: src/antq/util/maven.clj [hunk 2]
```diff
@@ -68,20 +68,42 @@
     (.setUsername (ensure-username-or-password username))
     (.setPassword (ensure-username-or-password password))))
 
+(defn- get-credential
+  []
+  (let [{:keys [username password]} (->> (u.lein/read-credential) vals first)]
```
[the remark this answers is not in the document]
Comment (blue0513) [reply src/antq/util/maven.clj]: Yes, that's correct

**Comment by blue0513** [comment]
@liquidz Thx for your review. I've responded to your review ! 
[`0f02594` (#237)](https://github.com/liquidz/antq/pull/237/commits/0f02594ab4f85f8dd5e578363395498a0f531a98)

File: src/antq/util/leiningen.clj (hunk 1 above)
[the remark this answers is not in the document]
Comment (blue0513) [reply src/antq/util/leiningen.clj:41]: However, I believe this implementation is quite common, so there is no need to be overly concerned about it.

File: src/antq/util/leiningen.clj [hunk 3]
```diff
@@ -17,6 +24,43 @@
...
+         (log/error (format "Could not decrypt credentials from %s" (str file)))
+         (log/error err)
+         (log/error "See `lein help gpg` for how to install gpg."))
+       (read-string out)))))
+
+(defn get-credential
+  [url]
+  (let [{:keys [username password]}
+        (some->> (credentials-fn)
+                 (filter (fn [[pattern, _]] (re-matches pattern url)))
```
[the remark this answers is not in the document]
Comment (liquidz) [reply src/antq/util/leiningen.clj:58]: > [pattern, _]

BTW, extra comma is not needed.

**Review by liquidz — CHANGES_REQUESTED** [review]
@blue0513 I'm sorry for the delayed response once again.
It was hard to find the time to test with the authenticated mvn repository, which has caused the delay.

File: src/antq/util/leiningen.clj (hunk 1 above)
[the remark this answers is not in the document]
Comment (blue0513) [reply src/antq/util/leiningen.clj:42]: Yes, I agree 👍

**Comment by blue0513** [comment]
@liquidz Big thx! I treated your review by this commit: 
 [`25bb320` (#237)](https://github.com/liquidz/antq/pull/237/commits/25bb3200d186418eec74ef7ba734f8c943115d06)

**Review by liquidz — APPROVED** [review]
@blue0513 Thanks for your fixes! LGTM

**Comment by liquidz** [comment]
@blue0513 Just released v2.7.1133 :)

**Comment by blue0513** [comment]
@liquidz  Thank you for your kind review & Great library !!!

**Pull request merged**